lang/cs/Org.Apache.REEF.Client/YARN/YARNREEFClient.cs

```diff
@@ -105,6 +106,40 @@ public IJobSubmissionResult SubmitAndGetJobStatus(JobRequest jobRequest)
             return application.FinalStatus;
         }
 
+        /// <summary>
+        /// Returns all the application reports running in the cluster.
+        /// GetApplicationAsync call is very expensive as it is trying 
+        /// fetch information about all the applications in the cluster.
+        /// </summary>
+        /// <returns></returns>
+        public async Task<List<IApplicationReport>> GetApplicationReports()
```

Comment:
If this method is called right after the app is submitted, the current app most possibly won't return until repeatedly retry.